sources/osgAnimation/BasicAnimationManager.js

```diff
@@ -160,30 +148,25 @@ define( [
             this.addAnimations( animations );
         },
 
+
+        // push all animations into the queue
         addAnimations: function ( animations ) {
 
-            // instanceAnimation added
             var instanceAnimationList = this._addAnimation( animations );
 
-            // compute a map and set a targetID for each InstanceChannel from
-            // instanceAnimationList
-            var newTargetAdded = Animation.registerChannelTargetID( instanceAnimationList, this._targets, this._targetsMap );
-
-            // register targetID per type
-            for ( var i = 0; i < newTargetAdded.length; i++ ) {
-                var type = newTargetAdded[ i ].type;
-                this._targetID.push( createTargetID( i, ResultType[ type ].create() ) );
-            }
-
+            // qeue them to assign target
+            Array.prototype.push.apply( this._animationsToRegister, instanceAnimationList );
             this._dirty = true;
+            this._registerAnimations();
```

Comment:
Yeah adding an animation tag the manager dirty, but registerAnimation can fix dirty to
false if we already has the upcateCallback collected. It's due to the fact that to
register animation we need to first collect data in the graph. So if not ready the update
will check if dirty and will fill collected data from the scene graph and be able to
finish to register animation